spec/data/zone_spec.rb

```diff
@@ -150,12 +150,38 @@ def stub_time(utc_now)
         expect( ZoneOffset.detect?('PST') ).to be false
       end
 
-      it 'returns true when given an offset' do
+      it 'returns true when given a numeric offset' do
         expect( ZoneOffset.detect?(10) ).to be true
       end
 
+      it 'returns true when given a one-digit hour offset' do
+        expect( ZoneOffset.detect?('1') ).to be true
+        expect( ZoneOffset.detect?('+1') ).to be true
+        expect( ZoneOffset.detect?('-1') ).to be true
+      end
+
+      it 'returns true when given a two-digit hour offset' do
+        expect( ZoneOffset.detect?('09') ).to be true
+        expect( ZoneOffset.detect?('+09') ).to be true
+        expect( ZoneOffset.detect?('-09') ).to be true
+      end
+
+      it 'returns true when given a four-digit offset' do
+        expect( ZoneOffset.detect?('0100') ).to be true
+        expect( ZoneOffset.detect?('+0100') ).to be true
+        expect( ZoneOffset.detect?('-1200') ).to be true
+      end
+
+      it 'returns true when preceded by a space' do
+        expect('August 9, 6:56 AM -10').to be true
```

Comment:
test is missing method to test on expectation.
`expect( ZoneOffset.detect?('August 9, 6:56 AM -10') ).to be true`

Reply:
Thanks, I had some unpushed fixes here. 😆